### ISSUE TITLE
Accessibility limitations for screen readers on add appointment screen

### ISSUE THREAD

**Issue by bookingwp** [bookingwp/woocommerce-appointments#857]
The add appointment screen has two controls that are not accessible for screen readers to use. The product drop down/ajax/filter control and the customer drop down/ajax/filter control. The jaws for windows screen reader cannot interact with these controls.

‌

A sample drop down / search /ajax control that is accessible for screen readers is the search box at [www.google.com](http://www.google.com)

‌

If you use the keyboard and tab into the search box and type atari, the ajax post works/filters results and the jaws for windows screen reader will read everything fine. You can down / up arrow through the filtered results and the screen reader reads everything well.

‌

Make sure to choose atari breakout! lol

‌

Can you please consider swapping out the current product / customer drop down/ajax/filter control you are using and replace it with another more accessible version?

‌

You could keep the control id the same so your existing code can work with it. May need adjusted slightly but keeping the control id the same should help allot.

‌

You may also be able to use bootstrap or simple CSS to style the new control to look similar to the non-accessible ones.

‌

If running windows, here is a free screen reader sighted developers use to test accessibility on their pages:

‌

[https://www.nvaccess.org/download/](https://www.nvaccess.org/download/)

‌

If on a mac, it comes built in with a screen reader called voice over. Just turn it on and you can use it to test pages.

‌

If on Linux, there are free open source screen readers that come bundled with certain versions of Linux that you can use for testing too.

‌

You can use the keyboard when using the screen reader but also can click on things with the mouse to more quickly jump to controls on the page to test them.

‌

For the product and customer controls, when you tab onto them with the keyboard, the screen reader should read the letters aloud as you type to filter the results, and then when you down arrow through the filtered results the screen reader should read each item so you can choose one.

‌

‌

Lastly, if these non-accessible drop down / ajax/filter controls are used anywhere else in the product would it be possible to just swap them out too and keep them all accessible?

**Comment by bookingwp** [comment]
**Željan Topić** (2021-10-01T14:55:55.433387+00:00):

We’re still evaluation this. At the moment this issue is present on the WooCommerce level, not just with our plugin.

**Comment by bookingwp** [comment]
**Željan Topić** (2025-10-27T16:09:15.897009+00:00):

Added with version 5.0.0-beta3
